dags/homelessness/sync_rap_data.py

```diff
@@ -0,0 +1,95 @@
+"""
+Pull data from RAP FTP and sync with ArcGIS Online
+"""
+import pandas as pd
+import ftplib
+import arcgis
+
+from airflow import DAG
+from airflow.operators.python_operator import PythonOperator
+from airflow.hooks.base_hook import BaseHook
+from datetime import datetime, timedelta
+
+def get_file(ftphost, ftpuser, ftppassword, filename):
+    ftp = ftplib.FTP(host = ftphost)
+    ftp.login(user = ftpuser, passwd = ftppassword)
+    ftp.cwd("Upload")
+    ftp.retrbinary('RETR %s' % filename, open("tmp/%s" % filename, 'wb').write)
+    
+def correct_file(filename):
+    filename = "tmp/%s" % filename
+    df = pd.read_csv(filename)
+    if "GeoLong" in df.columns:
+        df.loc[df["GeoLong"] > 0, "GeoLong"] *= -1
+    df["DummyField"] = 1
+    df.to_csv(filename, index = False)
+
+def update_arcgis(arcuser, arcpassword, arcfeatureid, filename):
+    filename = "tmp/%s" % filename
+    gis = arcgis.GIS("http://lahub.maps.arcgis.com", arcuser, arcpassword)
+    gis_item = gis.content.get(arcfeatureid)
+    gis_layer_collection = arcgis.features.FeatureLayerCollection.fromitem(gis_item)
+    gis_layer_collection.manager.overwrite(filename)
+
+default_args = {
+    "owner": "airflow",
+    "depends_on_past": False,
+    "start_date": datetime(2019, 11, 14),
+    "email": ["ian.rose@lacity.org", "hunter.owens@lacity.org", "brendan.bailey@lacity.org"],
+    "email_on_failure": True,
+    "email_on_retry": False,
+    "retries": 1,
+    "retry_delay": timedelta(days=1),
+}
+
+dag = DAG("rap-data", default_args=default_args, schedule_interval="@daily")
+
+
+def update_rap_data(**kwargs):
+    """
+    The actual python callable that Airflow schedules.
+    """
+    #Collecting data from FTP site
+    ftpconnection = BaseHook.get_connection("rap_ftp")
+    ftphost = ftpconnection.host
+    ftpuser = ftpconnection.login
+    ftppassword = ftpconnection.password
+    filename = kwargs.get("filename")
+    get_file(ftphost, ftpuser, ftppassword, filename)
+    
+    #Correcting issues with file
+    correct_file(filename)
+    
+    #Updating ArcGis
+    arcconnection = BaseHook.get_connection("rap_ftp")
```

Comment:
This needs a new name. We may be able to shoehorn the arcgis password/login into an FTP Connection type, or we may want to store this as an Airflow variable.

Reply:
Whoops - realized I assigned this the same name as the RAP connection. Thanks for pointing that out.